File: video-series-command/src/main/java/org/jesperancinha/video/command/commands/AddVideoSeriesCommand.java
package org.jesperancinha.video.command.commands;

import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import org.axonframework.modelling.command.TargetAggregateIdentifier;

import java.math.BigDecimal;

@Data
@Builder
@EqualsAndHashCode
@ToString
public class AddVideoSeriesCommand {

    @TargetAggregateIdentifier
    private String id;

    private String name;

    private Integer volumes;

    private BigDecimal cashValue;

    private String genre;

}

File: video-series-command/src/main/kotlin/org/jesperancinha/video/command/commands/AddVideoSeriesCommand.kt
package org.jesperancinha.video.command.commands

import org.axonframework.modelling.command.TargetAggregateIdentifier
import org.jesperancinha.video.core.data.Genre
import java.math.BigDecimal

data class AddVideoSeriesCommand(
    @TargetAggregateIdentifier
    val id: String? = null,
    val name: String? = null,
    val volumes: Int? = null,
    val cashValue: BigDecimal? = null,
    val genre: Genre? = null,
)